public/jsx/app.jsx

```diff
@@ -25,10 +27,23 @@ var App = React.createClass({
     var productId = router.getCurrentQuery().product;
 
     return {
+      error: AppStore.getError(),
       productId: productId,
     };
   },
 
+  componentDidMount: function() {
+    AppStore.addErrorListener(this.onError);
```

Comment:
one nice thing about redux (as chuck pointed out earlier) is that we won't have to explicitly set up (and test) all the event listeners.